homeassistant/components/mqtt/__init__.py

```diff
@@ -1247,6 +1265,18 @@ def device_info(self):
         return device_info_from_config(self._device_config)
 
 
+@websocket_api.websocket_command(
+    {vol.Required("type"): "mqtt/device/remove", vol.Required("device_id"): str}
+)
+@websocket_api.async_response
+async def websocket_remove_device(hass, connection, msg):
+    """Delete device."""
+    device_id = msg["device_id"]
+    dev_registry = await get_dev_reg(hass)
+    dev_registry.async_remove_device(device_id)
```

Comment:
This can raise a `KeyError`. We should also only delete devices that are part of MQTT.

Reply:
I think it's fine if it throws, this will result in an error response to the WS call?
Added check to only delete devices that are part of MQTT:
https://github.com/home-assistant/home-assistant/blob/e170a11547b716de8cfe46a99473f4dbfcb7f62b/homeassistant/components/mqtt/__init__.py#L1277-L1279